src/NServiceBus.Transport.RabbitMQ.TransportTests/Changing_concurrency.cs

```diff
@@ -0,0 +1,86 @@
+﻿namespace NServiceBus.Transport.RabbitMQ.TransportTests
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using NServiceBus.TransportTests;
+    using NUnit.Framework;
+
+    public class Changing_concurrency : NServiceBusTransportTest
+    {
+        [TestCase(TransportTransactionMode.ReceiveOnly)]
+        public async Task ImmediateRetries(TransportTransactionMode transactionMode)
+        {
+            var messageRetries = new TaskCompletionSource<MessageContext>();
+            int invocationCounter = 0;
+
+            await StartPump(async (context, ct) =>
+                {
+                    Interlocked.Increment(ref invocationCounter);
+
+                    await receiver.ChangeConcurrency(new PushRuntimeSettings(1), ct);
+                    await Task.Delay(500, ct);
+
+                    messageRetries.SetResult(context);
+                }, (_, ct) =>
+                {
+                    Assert.Fail();
+                    return Task.FromResult(ErrorHandleResult.RetryRequired);
+                },
+                transactionMode);
+
+            await SendMessage(InputQueueName);
+            await messageRetries.Task;
+            await Task.Delay(1000); // take some time to process any remaining message
```

Comment:
I think this could be lowered quite a bit but not sure we can get rid of some wait time completely.

Reply:
that's a good idea 👍